.github/workflows/style-guide.yml

```diff
@@ -0,0 +1,23 @@
+---
+name: Verify style guide compliance on pull request
```

Comment:
couple thoughts here...

1. why not do this as part of the pr.yml?
2. or, if you want it running in parallel (rather than blocking the generation of the PR preview in #650 if something fails teh vale check) could the filename be renamed to `pr-style-guide.yml` so it's obviously a PR check workflow?

Reply:
Yes for 2.

Reply:
cool, please proceed.